lib/src/main/java/com/icapps/architecture/utils/ext/LifecycleExt.kt

```diff
@@ -16,4 +16,29 @@ fun Lifecycle.addOnStopObserver(callback: () -> Unit) {
             this@addOnStopObserver.removeObserver(this)
         }
     })
+}
+
+/**
+ * Adds an observer to the specified Lifecycle, which will trigger the provided [callback] when the lifecycle hits ON_START and ON_STOP.
+ * The observer is registered immediately and gets cleaned up after the lifecycle hits ON_STOP OR ON_DESTROY,
+ * this means the callbacks can be called only once
+ */
+fun Lifecycle.addStartAndStopObservers(onStartCallback: () -> Unit, onStopCallback: () -> Unit) {
```

Comment:
Didn't we have an addOnPauseObserver too?

Reply:
Perhaps in another project

Reply:
Hm, it's exactly the same as the ON_STOP but with ON_PAUSE, I would add it too because it can be useful in many cases.